lib/ecl/ecl_sum_vector.c

```diff
@@ -58,6 +58,27 @@ ecl_sum_vector_type * ecl_sum_vector_alloc(const ecl_sum_type * ecl_sum){
     return ecl_sum_vector;
 }
 
+static void ecl_sum_vector_add_invalid_key(ecl_sum_vector_type * vector, const char * key) {
+  int_vector_append(vector->node_index_list, -1);
+  bool_vector_append(vector->is_rate_list, false);
+  stringlist_append_copy(vector->key_list, key);
+}
+
+
+
+ecl_sum_vector_type * ecl_sum_vector_alloc_layout_copy(const ecl_sum_vector_type * vector, const ecl_sum_type * ecl_sum) {
+  ecl_sum_vector_type * new_vector = ecl_sum_vector_alloc(ecl_sum);
+  for (int i=0; i < stringlist_get_size(vector->key_list); i++) {
+    const char * key = stringlist_iget(vector->key_list, i);
+    if (ecl_sum_has_general_var(ecl_sum, key))
+      ecl_sum_vector_add_key(new_vector, key);
+    else
+      ecl_sum_vector_add_invalid_key(new_vector, key);
+  }
+  return new_vector;
+}
+
```

Comment:
Do I understand correctly if `params_index < 0` then the element is invalid?

If so, should maybe have an `ecl_sum_vector_is_valid(const ecl_sum_vector_type * vec, int i)` that returns true if and only if `vec->node_index[i] < 0`?

Reply:
> Do I understand correctly if params_index < 0 then the element is invalid?

Yes

> should maybe have an ecl_sum_vector_is_valid(const ecl_sum_vector_type * vec, int i)

YAGNI??